tests/epyccel/test_epyccel_sign.py

```diff
@@ -0,0 +1,185 @@
+
+    assert f_pos_epyc() == f_pos()
+    assert f_neg_epyc() == f_neg()
+    assert f_nul_epyc() == f_nul()
+    assert f_nul_imag_epyc() == f_nul_imag()
+    assert f_real_nul_epyc() == f_real_nul()
+
+def test_sign_literal_int(language):
+    f_pos = mod.literal_int_pos
+    f_neg = mod.literal_int_neg
+    f_nul = mod.literal_int_nul
+    f_pos_epyc = epyccel(f_pos, language = language)
+    f_neg_epyc = epyccel(f_neg, language = language)
+    f_nul_epyc = epyccel(f_nul, language = language)
+
+    assert f_pos_epyc() == f_pos()
+    assert f_neg_epyc() == f_neg()
+    assert f_nul_epyc() == f_nul()
+
+def test_sign_literal_float(language):
+    f_pos = mod.literal_float_pos
+    f_neg = mod.literal_float_neg
+    f_nul = mod.literal_float_nul
+    f_pos_epyc = epyccel(f_pos, language = language)
+    f_neg_epyc = epyccel(f_neg, language = language)
+    f_nul_epyc = epyccel(f_nul, language = language)
+
+    assert f_pos_epyc() == f_pos()
+    assert f_neg_epyc() == f_neg()
+    assert f_nul_epyc() == f_nul()
+
+# Tests on arrays
+
+def test_sign_arr_int(language):
+    f_1d = mod.array_int_1d
+    f_2d = mod.array_int_2d
+
+    f_1d_epyc = epyccel(f_1d, language = language)
+    f_2d_epyc = epyccel(f_2d, language = language)
+
+    x1_1d = np.array([2, 0, 2, -13, 37, 42], dtype=np.int64)
+    x1_2d = np.array([[2, 0], [2, -13], [37, 42]], dtype=np.int64)
+    x2_1d = np.copy(x1_1d)
+    x2_2d = np.copy(x1_2d)
+
+    f_1d(x1_1d)
+    f_2d(x1_2d)
+    f_1d_epyc(x2_1d)
+    f_2d_epyc(x2_2d)
+
+    assert np.array_equal(x1_1d, x2_1d) and isinstance(x1_1d, type(x2_1d))
+    assert np.array_equal(x1_2d, x2_2d) and isinstance(x1_2d, type(x2_2d))
+
+def test_sign_arr_float(language):
+    f_1d = mod.array_float_1d
+    f_2d = mod.array_float_2d
+
+    f_1d_epyc = epyccel(f_1d, language = language)
+    f_2d_epyc = epyccel(f_2d, language = language)
+
+    x1_1d = np.array([0., 1., 2., -1.3, -3.7, -0.], dtype=np.float64)
+    x1_2d = np.array([[0., 1.], [2., -1.3], [-3.7, -0.]], dtype=np.float64)
+    x2_1d = np.copy(x1_1d)
+    x2_2d = np.copy(x1_2d)
+
+    f_1d(x1_1d)
+    f_2d(x1_2d)
+    f_1d_epyc(x2_1d)
+    f_2d_epyc(x2_2d)
+
+    assert np.array_equal(x1_1d, x2_1d) and isinstance(x1_1d, type(x2_1d))
+    assert np.array_equal(x1_2d, x2_2d) and isinstance(x1_2d, type(x2_2d))
+
+def test_sign_arr_complex(language):
+    f_1d = mod.array_complex_1d
+    f_2d = mod.array_complex_2d
+
+    f_1d_epyc = epyccel(f_1d, language = language)
+    f_2d_epyc = epyccel(f_2d, language = language)
+
+    x1_1d = np.array([0.+.1j, -2.-1.3j, -3.7j], dtype=np.complex64)
+    x1_2d = np.array([[0.+1.], [-2.-1.3j], [-3.7j]], dtype=np.complex64)
+    x2_1d = np.copy(x1_1d)
+    x2_2d = np.copy(x1_2d)
+
+    f_1d(x1_1d)
+    f_2d(x1_2d)
+    f_1d_epyc(x2_1d)
+    f_2d_epyc(x2_2d)
+
+    assert np.array_equal(x1_1d, x2_1d) and isinstance(x1_1d, type(x2_1d))
+    assert np.array_equal(x1_2d, x2_2d) and isinstance(x1_2d, type(x2_2d))
```

Comment:
Please check the return types in this file

Reply:
https://github.com/pyccel/pyccel/pull/1169/commits/0beda3e83578bfa9d98889ad2f1434fc97eacf17